File: 백준/Silver/3273. 두 수의 합/두 수의 합.py
import sys

n = int(sys.stdin.readline())

nums = list(map(int, sys.stdin.readline().split(" ")))

x = int(sys.stdin.readline())

startIdx = 0
endIdx = len(nums) - 1

count = 0

nums.sort()

while(startIdx < endIdx):
    sumNum = nums[startIdx] + nums[endIdx]
    if(sumNum > x):
        endIdx -= 1
    elif(sumNum < x):
        startIdx += 1
    else:
        count += 1
        startIdx += 1
        endIdx -= 1

print(count)

File: 백준/Silver/3273. 두 수의 합/두 수의 합.java
import java.util.*;
import java.io.*;


public class Main {

    private static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    private static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    private static StringTokenizer st;
    private static int n, x;
    private static int[] arr;
    private static int count = 0;

    private static void find() {
        int start = 0;
        int end = n - 1;
        while(start < end) {
            int sumNum = arr[start] + arr[end];
            if(sumNum == x) {
                count += 1;
                start += 1;
                end -= 1;
            }
            else if(sumNum > x) {
                end -= 1;
            }
            else if(sumNum < x) {
                start += 1;
            }
        }
    }

    public static void main(String[] args) throws IOException {

        n = Integer.parseInt(br.readLine());
        st = new StringTokenizer(br.readLine(), " ");
        arr = new int[n];

        for(int i = 0 ; i < n ; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        x = Integer.parseInt(br.readLine());

        Arrays.sort(arr);

        find();

        bw.write(Integer.toString(count));
    

        bw.flush();

        br.close();
        bw.close();
    }
}